### PR TITLE
Story/14868

### PR DESCRIPTION
**Pull request by kenvac**
Adds new config on stock.picking.type to not allow pickings with destination location block on batch

### CODE REVIEW

File: addons/udes_stock/models/stock_picking.py [hunk 1]
```diff
@@ -2463,13 +2463,16 @@ def search_for_pickings(self, picking_type_id, picking_priorities, limit=1, doma
...
                 search_domain.append(("u_location_category_id", "child_of", categories.ids))
 
-        search_domain.extend(
-            [
-                ("picking_type_id", "=", picking_type_id),
-                ("state", "=", "assigned"),
-                ("batch_id", "=", False),
-            ]
-        )
+        extra_filter = [
```
Comment (armandcela) [inline addons/udes_stock/models/stock_picking.py:2466]: To be consistent in the way search_domain have been extended i would suggest not to use a new variable extra_filter, but directly extend search_domain if picking_type.u_batch_dest_loc_not_allowed

File: addons/udes_stock/models/stock_picking_type.py [hunk 2]
```diff
@@ -429,6 +429,12 @@ class StockPickingType(models.Model):
         default=False,
         help="If set, the user will be shown the list of all tasks during the picking workflow",
     )
+    u_batch_dest_loc_not_allowed = fields.Boolean(
+        string="Do not allow pickings with block destination location",
```
Comment (PeterC10) [inline addons/udes_stock/models/stock_picking_type.py:433]: This string is too long, won't display nicely on UI. Could you simplify the title? We always have the help string for a more in-depth explanation.

Also please remember to capitalise the string, capitalizemytitle.com is a useful tool for this. :)

**Comment by PeterC10** [comment]
General nit: Could you shorten the title of the first commit? It should be no more than 72 characters based on our CR checklist.

I think you could also fixup the second commit into the first.

File: addons/udes_stock/models/stock_picking_type.py (hunk 2 above)
Comment (kenvac) [reply addons/udes_stock/models/stock_picking_type.py:433]: How about "No blocked location pickings" sounds to you?

Comment (PeterC10) [reply addons/udes_stock/models/stock_picking_type.py:433]: I think we need to include the fact it's for destination locations, how about `No Blocked Dest. Location Pickings`? Still a bit long but cut down a fair bit from before.

Might be worth getting another opinion if unsure.

**Review by armandcela — APPROVED** [review]

**Review by PeterC10 — APPROVED** [review]
LGTM

**Pull request merged**